HackIllinois/Models/APIModels/HIAPIEvent.swift

```diff
@@ -11,32 +11,37 @@
 //
 
 import Foundation
+import APIManager
 
-struct HIAPIEvent: Codable {
-    typealias Contained = HIAPIReturnDataContainer<HIAPIEvent>
+struct HIAPIEventsContainer: Decodable, APIReturnable {
+    let events: [HIAPIEvent]
+}
+
+struct HIAPIEvent: Codable, APIReturnable {
+    typealias Contained = HIAPIEventsContainer
 
     private enum CodingKeys: String, CodingKey {
-        case id
         case name
         case info = "description"
         case end = "endTime"
         case start = "startTime"
-        case locations
+        case eventType
+        case latitude
+        case longitude
+        case locationDescription
+        case sponsor
     }
 
     var favorite = false
-    var id: Int16
     var name: String
     var info: String
-    var end: Date
-    var start: Date
-    var locations: [HIAPILocationReference]
-
-    struct HIAPILocationReference: Codable {
-        var id: Int16
-        var eventId: Int16
-        var locationId: Int16
-    }
+    var end: Double
```

Comment:
why is this a double?

Reply:
because TimeInterval accepts a double when converting to Date, but Int/Int16 should also work fine

Reply:
What is the number coming from the API? Time since 1970?

Reply:
then this should be type `Date` where the APIReturnable `init(from:)` method is overriden to provide a JSONDecoder that uses the proper date decoding strategy, see docs [here](https://developer.apple.com/documentation/foundation/jsondecoder/datedecodingstrategy)

Reply:
ah ok, currently it loops through apievents and makes them events (which have type `Date`)